### ISSUE TITLE
DataTypeException when org.jooq.Meta accesses PostgreSQL array column with default expression

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#8468]
Create this table in PostgreSQL:

```sql
CREATE TABLE t(
  a bigint[] DEFAULT '{}'::bigint[]
);
```

Now access it via `org.jooq.Meta`:

```java
List<Schema> schemas = ctx.meta().getSchemas("public");
for (var schema : schemas) {
    System.out.println(schema.getTables());
}
```

The following exception is thrown:

```
org.jooq.exception.DataTypeException: Cannot convert from '{}'::bigint[] (class java.lang.String) to class [Ljava.lang.Long;
	at org.jooq.tools.Convert$ConvertAll.fail(Convert.java:1166)
	at org.jooq.tools.Convert$ConvertAll.from(Convert.java:1055)
	at org.jooq.tools.Convert.convert0(Convert.java:322)
	at org.jooq.tools.Convert.convert(Convert.java:314)
	at org.jooq.tools.Convert.convert(Convert.java:385)
	at org.jooq.impl.DefaultDataType.convert(DefaultDataType.java:769)
	at org.jooq.impl.DSL.val(DSL.java:20555)
	at org.jooq.impl.DSL.inline(DSL.java:20142)
	at org.jooq.impl.MetaImpl$MetaTable.init(MetaImpl.java:810)
	at org.jooq.impl.MetaImpl$MetaTable.<init>(MetaImpl.java:508)
	at org.jooq.impl.MetaImpl$MetaSchema.getTables(MetaImpl.java:408)
```

We might need to resort to the parser to fix this thoroughly (#8469, similar to #7849), but a quick fix is to simply ignore the default if it cannot be converted to the target type.

**Comment by studouglas** [comment]
Will the fix from `3dcba4e` go into the 3.11.12 release, or is it waiting for 3.12.0?

Our tests rely on `meta.getTables()` so it's currently blocking us from upgrading to 3.11 (from 3.10). If there are any workarounds in the meantime I'm all ears :)

**Comment by lukaseder** [comment]
> Will the fix from 3dcba4e go into the 3.11.12 release, or is it waiting for 3.12.0?

Will go to 3.11.12 (#8523). I forgot to create an issue. Created one now.

> Our tests rely on meta.getTables() so it's currently blocking us from upgrading to 3.11 (from 3.10). If there are any workarounds in the meantime I'm all ears :)

You can always patch your version of jOOQ. I cannot think of any other workaround.
